nixos/modules/services/mail/sympa.nix

```diff
@@ -0,0 +1,524 @@
+      { name = cfg.user;
+        description = "Sympa mailing list manager user";
+        group =  cfg.group;
+        home = cfg.dataDir;
+        createHome = true;
+        isSystemUser = true;
+      };
+
+    users.groups =
+      optional (cfg.group == "sympa")
+      { name = cfg.group;
+      };
+
+    warnings = optional (cfg.database.password != "")
+      ''config.services.sympa.database.password will be stored as plaintext
+        in the Nix store. Use database.passwordFile instead.'';
+
+    # Create database passwordFile default when password is configured.
+    services.sympa.database.passwordFile =
+      (mkDefault (toString (pkgs.writeTextFile {
+        name = "sympa-database-password";
+        text = cfg.database.password;
+      })));
+
+    systemd.services.sympa = {
+      description = "Sympa mailing list manager";
+
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+      wants = sympaSubServices;
+      before = sympaSubServices;
+      restartTriggers = [ mainConfig cfg.database.passwordFile ];
+      serviceConfig = sympaServiceConfig "sympa_msg";
+
+      preStart = ''
+        umask 0077
+        mkdir -p ${dataDir}/etc
+        mkdir -p ${dataDir}/spool
+        mkdir -p ${dataDir}/list_data
+        mkdir -p ${dataDir}/arc
+        mkdir -p ${dataDir}/bounce
+
+        cp -f ${mainConfig} ${dataDir}/etc/sympa.conf
+        DBPASS="$(head -n1 ${cfg.database.passwordFile})"
+        if [ -n "$DBPASS" ]; then
+            sed -e "s,#dbpass#,$DBPASS,g" \
+                -i ${dataDir}/etc/sympa.conf
+        else
+            sed -e "/db_passwd.*#dbpass#/d" \
+                -i ${dataDir}/etc/sympa.conf
+        fi
+
+        ${concatStringsSep "\n" (flip mapAttrsToList cfg.domains (fqdn: domain:
+        ''
+          mkdir -p -m 750 ${dataDir}/etc/${fqdn}
+          cp -f ${robotConfig fqdn domain} ${dataDir}/etc/${fqdn}/robot.conf
+          mkdir -p -m 750 ${dataDir}/list_data/${fqdn}
+        ''))}
+
+        cp -f ${virtual} ${dataDir}/virtual.sympa
+        cp -f ${transport} ${dataDir}/transport.sympa
+        cp -f ${listAliases} ${dataDir}/etc/list_aliases.tt2
+
+        touch ${dataDir}/sympa_transport
+
+        ${optionalString (cfg.mta.type == "postfix") ''
+          ${pkgs.postfix}/bin/postmap hash:${dataDir}/virtual.sympa
+          ${pkgs.postfix}/bin/postmap hash:${dataDir}/transport.sympa
+        ''}
+        ${pkg}/bin/sympa_newaliases.pl
+
+
+        cp -af ${pkg}/static_content ${dataDir}/
+        # Yes, wwsympa needs write access to static_content..
+        chmod -R u+w,a+X ${dataDir}/static_content/
+        # Make static_content accessible to nginx
+        chmod o+X ${dataDir}
+
+        ${pkg}/bin/sympa.pl --health_check
+      '';
+    };
+    systemd.services.sympa-archive = {
+      description = "Sympa mailing list manager (archiving)";
+      bindsTo = [ "sympa.service" ];
+      restartTriggers = [ mainConfig cfg.database.passwordFile ];
```

Comment:
Should be able to remove these `restartTriggers` too for the same reason

Reply:
Yep, works fine, thanks!